greedy_algorithms/gale_shapley.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file
+ * @brief Stable Marriage Problem implementation
+ * @details
+ * This implementation utilizes the Gale-Shapley algorithm to find stable matches
+ * between men and women based on their preferences.
+ *
+ * **Stable Marriage Problem** aims to find a stable matching between two equally sized 
+ * sets of elements given an ordinal preference for each element. The algorithm was
+ * introduced by David Gale and Lloyd Shapley in 1962.
+ *
+ * @author [B Karthik](https://github.com/BKarthik7)
+ */
+
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <algorithm>
+using namespace std;
+
+void stableMatch(const vector<vector<int>>& women_prefs, const vector<vector<int>>& men_prefs, int n) {
```

Comment:
```suggestion
void stable_match(const vector<vector<int>>& women_prefs, const vector<vector<int>>& men_prefs, int n) {
```

prefer snake_case to camelCase